### ISSUE TITLE
During the collab, an error message pops up

### ISSUE THREAD

**Issue by LiraBrik** [excalidraw/excalidraw#6221]
During the collab, an error message pops up "Couldn't save to the backend database, the canvas seems to be too big. You should save the file locally to ensure you don't lose your work.". As a result, there is no way to continue the collab, the constant appearance of an error window interferes. Saving to a local file and refreshing the page does not help.

**Comment by dwelle** [comment]
Correct, saving the file locally is only to solve the problem of "not losing data". It won't solve the size limit issue on the server. Refreshing still reconnects to the same collaboration room.

The constant message is not great, we should figure that out. Perhaps show once, and then show a status icon on canvas instead.

**Comment by doa1996** [comment]
I understand the reason behind this error and for me it shows up at the correct moment – when excalidraw (free browser app) can't handle more data and I keep losing connection with my students – so the message itself isn't the key part of the problem. 

My current workaround is to use excalidraw in incognito and, after the error happens: export a picture → stop session → clear canvas → exit (and tell my student to do so too) → reopen and start a new session in new incognito window → send my student a new link.
(Maybe not every step is necessary, but I don't feel like checking everything during work).
Works, but it's less than ideal to do that every ~50–60 minutes, especially that my lessons are most often a clock hour long, just enough for it to break but not be worth of restarting. 

From my perspective even a gauge telling me how 'full' is my current board would be helpful – this way I can anticipate the break and move on with another exercise onto a new session.
This could be also solved by allowing me to clear canvas during the shared session – show a warning that I better export what currently is on the board, then you can free your server from my data and I can go on with my lesson.

Meanwhile any more convenient workaround would be appreciated, even if one of them would be to buy the product (but if I'm to pay, it would absolutely need to work), I like excalidraw and want to keep using it.

**Comment by dwelle** [comment]
Hey @doa1996 — do you create a new collab room for every hour-long class? In other words, you are able to fill the canvas above limit (1MB) in one teaching session? If so, that is very strange, as only the shapes count towards the limit (images do not count).

As for a gauge, if you open the stats for nerds (right-click on canvas), the `storage total size` is a rough estimate.

**Comment by LiraBrik** [comment]
@dwelle I would like to thank you for the support of the Excalidraw community. This is a very useful and interesting project.
With the help of clear instructions posted on the site, I learned how to work with github, I can run Excalidraw locally. It's a good experience ) To work, I need an online whiteboard, I used Miro before, but now I'm looking for an alternative. I really liked both the design and the concept of Excalidraw - that I can customize the board to my needs. Various moments appear during the collaboration - CTRL+Z doesn't work, after 50-60 minutes the interlocutor stops seeing what I'm writing, the screenshot is not saved to the clipboard as PNG. 
Now I have cloned Excalidraw from github to the Ubuntu 20.04 server to try to make changes for personal use. The path pointed to ../public/index.html . But the board does not work, an error appears: <% if (process.env.REACT_APP_DEV_DISABLE_LIVE_RELOAD==="true" ) { %> <% } %> Perhaps you can offer some recommendations how to run Excalidraw on Ubuntu Server?

**Comment by LiraBrik** [comment]
Good afternoon! For people who liked the Excalidrow, it really lacks just a little bit of memory to work for 1 hour or 1 hour 20 minutes. I would suggest increasing "export const MAX_ALLOWED_FILE_BYTES = 2 * 1024 * 1024" by 10 times, or at least 5 times. And reduce the storage time "export const DELETED_ELEMENT_TIMEOUT = 24 * 60 * 60 * 1000". Since the program is open source, in principle I can make these changes. Of course, I won't be doing this anytime soon, unless within my own server when I get my hands on it.

**Comment by QiyuanChen02** [comment]
@dwelle To fix this, I'm thinking of only showing the maximum file exceeded error once as a popup and then having an icon at the bottom to state the error again on hover. This UI change should make it useable during the collab even if the data can no longer be saved to firebase, but the error icon on the bottom should still remind the user that the is a problem with the save, like this:

![image](https://github.com/excalidraw/excalidraw/assets/82120596/6aabba17-924d-4b5b-b510-93b0fd4daa86)

The suggested method will only try to save to firebase once, and when it finds maximum size exceeded then it will no longer try to save.

![image](https://github.com/excalidraw/excalidraw/assets/82120596/6c3c33b6-96c3-4aef-8a96-b467047363c6)

Will this implementation be satisfactory, or does not saving the collab room to Firebase result in any other problems in the rest of the code. If you think this implementation is fine, then I'm happy to try and do it.

**Comment by LiraBrik** [comment]
Great idea! I personally no longer have this error, since I posted the code
on my VDS. But I still haven’t figured out what to do with strl+z during
the session.

вс, 28 янв. 2024 г. в 19:59, QiyuanChen02 ***@***.***>:

**Comment by bioagbob** [comment]
Noting same/similar issue reported in: 
- #7204

**Comment by mknasiruddin** [comment]
Hello! i would like to work on it, shall i open a PR ?

**Comment by brovaset** [comment]
Hi! I'd like to take this on.

I've reviewed the issue and planned the implementation details:

State Management: Add a flag (hasSeenCollabSaveError / session state) to track whether the backend save error modal has already been triggered.

UI/UX Flow: Limit the modal popup to display once upon error, then suppress subsequent popups and render a non-intrusive error status indicator on the canvas/toolbar instead.

Testing: Include unit/UI tests covering both the initial modal trigger and the canvas status indicator state.

Could you please assign this to me so I can submit a PR?

**Comment by AndresBallares** [comment]
Hello.  I will like to claim and work on this issue.  Thanks
